tests/buildkite/build-containers.sh

```diff
@@ -20,7 +20,7 @@ case "${container}" in
   cpu)
     ;;
 
-  gpu)
+  gpu | gpu_dev_ver)
```

Comment:
It looks to me like this and `gpu_build_centos7` are identical... could those cases be consolidated to just `gpu*`?

(pending the outcome of this other discussion: https://github.com/dmlc/xgboost/pull/10351#discussion_r1622403527)